tools/rostopic/src/rostopic/__init__.py

```diff
@@ -788,7 +797,9 @@ def _rostopic_echo(topic, callback_echo, bag_file=None, echo_all_topics=False):
                     index = submsg_class.__slots__.index(field)
                     type_information = submsg_class._slot_types[index]
                     if fields:
-                        submsg_class = roslib.message.get_message_class(type_information)
+                        submsg_class = roslib.message.get_message_class(type_information.split('[')[0])
+                        if not submsg_class:
```

Comment:
`type_information` for an array element will contain the array subscript, e.g. `field[2]`. 
However, this type string doesn't have a valid message class. The type is named `field` in this case. Hence, we need to drop the brackets.